fdbserver/workloads/ClogRemoteTlog.actor.cpp

```diff
@@ -0,0 +1,161 @@
+		lagMeasurementFrequencySec = getOption(options, "lagMeasurementFrequencySec"_sr, 5);
+		clogInitDelaySec = getOption(options, "clogInitDelaySec"_sr, 5);
+		clogDurationSec = getOption(options, "clogDurationSec"_sr, 5);
+		lagThresholdSec = getOption(options, "lagThresholdSec"_sr, 5);
+	}
+
+	Future<Void> setup(const Database& db) override { return Void(); }
+
+	Future<Void> start(const Database& db) override {
+		if (g_network->isSimulated() && enabled) {
+			return timeout(reportErrors(workload(this, db), "ClogRemoteTLogError"), testDuration, Void());
+		}
+		return Void();
+	}
+
+	Future<bool> check(const Database& db) override { return true; }
+
+	void getMetrics(std::vector<PerfMetric>& m) override {}
+
+	ACTOR static Future<Void> measureMaxSSLagSec(ClogRemoteTLog* self, Database db) {
+		StatusObject status = wait(StatusClient::statusFetcher(db));
+		StatusObjectReader reader(status);
+		StatusObjectReader cluster;
+		StatusObjectReader processMap;
+		if (!reader.get("cluster", cluster)) {
+			TraceEvent("NoCluster");
+			return Void();
+		}
+		if (!cluster.get("processes", processMap)) {
+			TraceEvent("NoProcesses");
+			return Void();
+		}
+		double maxSSLagSec{ -1 };
+		for (auto p : processMap.obj()) {
+			StatusObjectReader process(p.second);
+			if (process.has("roles")) {
+				StatusArray roles = p.second.get_obj()["roles"].get_array();
+				for (StatusObjectReader role : roles) {
+					ASSERT(role.has("role"));
+					if (role.has("data_lag")) {
+						ASSERT(role["role"].get_str() == "storage");
+						auto dataLag = role["data_lag"].get_obj();
+						ASSERT(dataLag.contains("seconds"));
+						ASSERT(dataLag.contains("versions"));
+						TraceEvent("SSDataLag")
+						    .detail("Process", p.first)
+						    .detail("Role", role["role"].get_str())
+						    .detail("SecondLag", dataLag["seconds"].get_value<double>())
+						    .detail("VersionLag", dataLag["versions"].get_int64());
+						maxSSLagSec = std::max(maxSSLagSec, dataLag["seconds"].get_value<double>());
+					}
+				}
+			}
+		}
+		TraceEvent("MaxSSDataLag").detail("SecondLag", maxSSLagSec).detail("SecondThreshold", self->lagThresholdSec);
+		// TODO (praza): Uncomment this assert when gray failure detection is improved to automatically fix this issue
+		// ASSERT(maxSSLagSec < self->lagThresholdSec);
+		return Void();
+	}
+
+	ACTOR static Future<std::vector<IPAddress>> getRemoteSSIPs(Database db) {
+		state std::vector<IPAddress> ret;
+		Transaction tr(db);
+		tr.setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+		tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+		tr.setOption(FDBTransactionOptions::LOCK_AWARE);
+		std::vector<std::pair<StorageServerInterface, ProcessClass>> results =
+		    wait(NativeAPI::getServerListAndProcessClasses(&tr));
+		for (auto& [ssi, p] : results) {
+			if (ssi.locality.dcId().present() && ssi.locality.dcId().get() == g_simulator->remoteDcId) {
+				ret.push_back(ssi.address().ip);
+			}
+		}
```

Comment:
These txn read should be put into a `try...catch` block so that when encountering errors, the read can be retried.